controllers/spacerequest/spacerequest_controller.go

```diff
@@ -399,6 +412,125 @@ func (r *Reconciler) deleteExistingSubSpace(logger logr.Logger, subSpace *toolch
 	return true, nil
 }
 
+func (r *Reconciler) ensureSecretForProvisionedNamespaces(logger logr.Logger, memberClusterWithSpaceRequest cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, subSpace *toolchainv1alpha1.Space) error {
+	if len(subSpace.Status.ProvisionedNamespaces) == 0 {
+		logger.Info("provisioned namespaces not available yet...")
+		return nil
```

Comment:
Right, in theory when the space controller updates `.Status.ProvisionedNamespaces` of the subSpace , a reconcile is triggered in the space request controller which will then detect that the list of provisioned namespaces is not empty anymore.

Reply:
yes, exactly. Since we watch the Space CRD without any predicate:
```go
b := ctrl.NewControllerManagedBy(mgr).
	For(&toolchainv1alpha1.SpaceRequest{}).
	Watches(&source.Kind{Type: &toolchainv1alpha1.Space{}},
		handler.EnqueueRequestsFromMapFunc(MapSubSpaceToSpaceRequest()),
	)
```
the the controller gets reconciled as soon as the Space.Status is updated with the information about the provisioned namespaces. There is no reason for reaching out to the namespaces and getting the token of the SAs if they are not fully provisioned.